### PR TITLE
Closes #64

### LINKED ISSUE #64: `prob` method clean up
See https://github.com/Roche/crmPack/blob/b4af53b6585e8a83fc1765cfc9e074732dbd3bfe/R/Model-methods.R#L140

To do:

- [x] Revise documentation
  - [x] RStudio outline section dividers (use `# function or class name ----`)
  - [x] Add lifecycle badge (as a default take "stable" for existing functions or classes)
  - [x] Replace `##'` by `#'`
  - [x] Add to `pkgdown.yml`
  - [x] Remove file history and the timestamp in comments on top of file
- [x] Add unit tests
  - [x] For classes:
    - [x] Test default constructor
    - [x] For reference classes: test methods
  - [x] For functions: as usual
- [x] Refactor as needed
  - [x] Code styling (indentation etc.) - install package `styler` and use corresponding add-in
  - [x] Keep externally used names so that we don't break downstream code
  - [x] Remove `keywords` tags
  - [x] For functions or methods:
    - [x] Add assertions for arguments using `checkmate` functions
    - [x] If it is too long, break up into pieces (rule of thumb: maximum 50 lines of code)
    - [x] Consider renaming internal variables etc. if needed
    - [x] Avoid explicit use of `return`
  - [x] For classes:
    - [x] move function definitions outside if possible (e.g. move validation function as `validity_subjects()` to `Data-validity.R` and have corresponding test file `test-Data-validity.R`)
    - [x] In `setClass` function, just use the `slots` instead of outdated `representation`
    - [x] Remove things like `validObject(.Data())`

### PR DESCRIPTION
**Pull request by wwojciech**
Closes #64 + minor character change in help pages for some other methods/classes.

### CODE REVIEW

File: R/Model-methods.R [hunk 1]
```diff
@@ -10,8 +10,8 @@ NULL
 #' @param prob (`number`)\cr the toxicity probability which is targeted.
 #' @param model (`Model` or `ModelTox`)\cr the model for single agent dose
 #'   escalation or pseudo DLE (dose-limiting events)/toxicity model.
-#' @param samples (`Samples`)\cr the samples that will be used to compute the
-#'   resulting dose by `model@dose` function.
+#' @param samples (`Samples`)\cr the samples of model's parameters that will be
+#'   used to compute the resulting dose by `model@dose` function.
```
Comment (danielinteractive) [inline R/Model-methods.R:14]: Actually "doses" would be more accurate (multiple samples so multiple doses)

File: R/Model-methods.R [hunk 2]
```diff
@@ -116,114 +116,122 @@ setMethod(
...
-                                  "dose")
-              ## now call the function with dose and with
-              ## the arguments taken from the samples
-              ret <- do.call(probFun,
-                             c(list(dose=dose),
-                               samples@data[argNames]))
-              ## return the resulting vector
-              return(ret)
-          })
+#' Computing Toxicity Probabilities for a Given Dose, Model and the Samples
```
Comment (danielinteractive) [inline R/Model-methods.R:121]:
```suggestion
#' Computing Toxicity Probabilities for a Given Dose, Model and Samples
```

File: R/Model-methods.R [hunk 3]
```diff
@@ -116,114 +116,122 @@ setMethod(
...
-              ret <- do.call(probFun,
-                             c(list(dose=dose),
-                               samples@data[argNames]))
-              ## return the resulting vector
-              return(ret)
-            })
+#' @rdname prob
+#'
+#' @description Compute toxicity probabilities for a given dose, a given single
+#'   agent dose escalation model, and the samples.
```
Comment (danielinteractive) [inline R/Model-methods.R:151]:
```suggestion
#'   agent dose escalation model, and samples.
```

File: R/Model-methods.R [hunk 4]
```diff
@@ -116,114 +116,122 @@ setMethod(
...
+setMethod(
+  f = "prob",
+  signature = signature(
+    dose = "numeric",
+    model = "Model",
+    samples = "Samples"
+  ),
+  definition = function(dose, model, samples, ...) {
+    prob_fun <- model@prob
+    # Which arguments, besides the `dose`, does `prob_fun` need?
```
Comment (danielinteractive) [inline R/Model-methods.R:166]:
```suggestion 
```

File: R/Model-methods.R [hunk 5]
```diff
@@ -116,114 +116,122 @@ setMethod(
...
+setMethod(
+  f = "prob",
+  signature = signature(
+    dose = "numeric",
+    model = "ModelTox",
+    samples = "Samples"
+  ),
+  definition = function(dose, model, samples, ...) {
+    prob_fun <- model@prob
+    # Which arguments, besides the `dose`, does `prob_fun` need?
```
Comment (danielinteractive) [inline R/Model-methods.R:192]: Same here

File: R/Model-methods.R [hunk 6]
```diff
@@ -116,114 +116,122 @@ setMethod(
...
+setMethod(
+  f = "prob",
+  signature = signature(
+    dose = "numeric",
+    model = "ModelTox",
+    samples = "missing"
+  ),
+  definition = function(dose, model, ...) {
+    prob_fun <- model@prob
+    # Which arguments, besides the `dose`, does `prob_fun` need?
```
Comment (danielinteractive) [inline R/Model-methods.R:220]: And here

File: R/Model-methods.R [hunk 7]
```diff
@@ -116,114 +116,122 @@ setMethod(
...
+    prob_args <- setdiff(formalArgs(prob_fun), "dose")
+    if (!all(prob_args %in% slotNames(model))) {
+      stop("Arguments to `model@prob` (except `dose`) must be in the `model`")
+    }
+    prob_args_values <- sapply(
+      prob_args,
+      function(a) slot(model, a),
```
Comment (danielinteractive) [inline R/Model-methods.R:227]: Should not need another function definition here, can just appropriately pass the model to named first argument

File: examples/Model-method-dose.R [hunk 8]
```diff
@@ -6,7 +6,7 @@ my_data <- Data(
   doseGrid = c(0.1, 0.5, 1.5, 3, 6, seq(from = 10, to = 80, by = 2))
 )
 
-# Initialize a model, e.g. `LogisticLogNormal`.
+# Initialize a model, e.g. 'LogisticLogNormal'.
```
Comment (danielinteractive) [inline examples/Model-method-dose.R:9]: This (apostrophe vs backtick in comments) I would not be too picky about but ok

**Review by danielinteractive — APPROVED** [review]
Thanks Wojciech, only Minor comments inside, good to merge afterwards

File: R/Model-methods.R (hunk 7 above)
Comment (wwojciech) [reply R/Model-methods.R:227]: Sure, thanks for the hint!

**Pull request merged**